lms/djangoapps/grades/api/v1/views.py

```diff
@@ -113,6 +113,39 @@ class CourseEnrollmentPagination(CursorPagination):
     ordering = 'created'
 
 
+class PaginatedAPIView(APIView):
+    """
+    An `APIView` class enhanced with the pagination methods of `GenericAPIView`.
```

Comment:
Are you proposing this as a best practice moving forward - that is, use `PaginatedAPIView` instead of `GenericAPIView` when not using a `Serializer`?  If so, we should move this into a more common library.

On the other hand, this direction is having us fork our own Pagination implementation from DRF's.

CC @ormsbee if he has any perspective on this.

Reply:
I don't know.  I think it's the fact that we're not defining a `queryset` attribute forcing us to deviate from DRF's opinion - this code still uses a serializer in a few places, albeit in a more explicit way than simply declaring a serializer class as an attribute of the view class.  I'm not sure if DRF defines a "right way" to set up views/pagination/serializers when the data you want to serve doesn't come directly from a django query set.

Here's what the DRF docs have to say about custom pagination, FWIW: https://www.django-rest-framework.org/api-guide/pagination/#custom-pagination-styles

Anyway, if a `PaginatedAPIView` has enough utility and re-usability to warrant it, then this could be moved into edx-drf-extensions.  Or maybe we could create a PR against DRF itself to get these pagination methods added to `APIView`?

Reply:
A PR or github issue against DRF - could at least give us some indication on how they recommend we proceed.

The [other 4 APIs that have excluded themselves from Swagger docs](https://github.com/edx/edx-platform/search?q=exclude_from_schema&unscoped_q=exclude_from_schema) have the same issue (and depend on `GenericAPIView`).  @douglashall Did you happen to take a deeper look at those APIs at that time to see how to deal with this?